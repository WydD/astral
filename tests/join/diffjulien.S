deviceId,value,T
A,1,1
A,2,1
A,3,1
A,4,1
B,1,2
B,2,2
A,1,3
A,2,3
A,3,3
A,5,3
B,1,4
B,2,4
B,3,5

