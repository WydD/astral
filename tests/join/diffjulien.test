<?xml version="1.0" encoding="UTF-8"?>
<astral xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="http://astral.ligforge.imag.fr/schema/query.xsd">
	<query>
        <streamer stype="Rsu">
            <sigma condition="value == 'null' || oldvalue == 'null'">
                <join defaultOuter="null" outer="full" condition="value==oldvalue">
                    <pi removeAttributes="T">
                        <window partitionedBy="deviceId">
                            <description wtype="B"/>
                            <source id="S"/>
                        </window>
                    </pi>
                    <pi removeAttributes="T">
                        <rho>
                            <rename from="value" to="oldvalue"/>
                            <domain dtype="previous">
                                <window partitionedBy="deviceId">
                                    <description wtype="B"/>
                                    <source id="S"/>
                                </window>
                            </domain>  
                        </rho>
                    </pi>
                </join>
            </sigma>
        </streamer>
	</query>
	<sources>
		<source file="tests/join/diffjulien.S" id="S" impl="FileStreamImpl"/>
	</sources>
</astral>
