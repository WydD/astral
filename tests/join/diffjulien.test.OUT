T,oldvalue,value,deviceId
1,null,1,A
1,null,2,A
1,null,3,A
1,null,4,A
2,null,1,B
2,null,2,B
3,4,null,A
3,null,5,A
5,1,null,B
5,2,null,B
5,null,3,B

